esp-wifi/src/wifi/mod.rs

```diff
@@ -645,28 +645,35 @@ unsafe extern "C" fn recv_cb(
     len: u16,
     eb: *mut c_types::c_void,
 ) -> esp_err_t {
-    let res = critical_section::with(|cs| {
-        let mut queue = DATA_QUEUE_RX.borrow_ref_mut(cs);
-        if queue.is_full() {
-            error!("RX QUEUE FULL");
-            esp_wifi_internal_free_rx_buffer(eb);
-            include::ESP_ERR_NO_MEM as esp_err_t
-        } else {
-            let packet = EspWifiPacketBuffer { buffer, len, eb };
-            unwrap!(queue.enqueue(packet));
+    critical_section::with(|cs| {
+        let packet = EspWifiPacketBuffer { buffer, len, eb };
 
-            #[cfg(feature = "embassy-net")]
-            embassy::RECEIVE_WAKER.wake();
+        match DATA_QUEUE_RX.borrow_ref_mut(cs).enqueue(packet) {
+            Ok(_) => {
+                #[cfg(feature = "embassy-net")]
+                embassy::RECEIVE_WAKER.wake();
 
-            include::ESP_OK as esp_err_t
-        }
-    });
+                include::ESP_OK as esp_err_t
+            }
 
-    res
+            _ => {
+                debug!("RX QUEUE FULL");
+                include::ESP_ERR_NO_MEM as esp_err_t
```

Comment:
we should still do `esp_wifi_internal_free_rx_buffer(eb);`

Reply:
It's done by `EspWifiPacketBuffer::drop`, or at least that is the intention.

Reply:
oh I see .... I should stop reviewing code via GH UI

Reply:
And I should probably start describing my changes in more detail so you have the answers before you need to ask 😅